### PR TITLE
fix: in case the resource id is not present 

### PR DESCRIPTION
**Pull request by AmirthavalliG**
fix: in case the resource id is not present …
errorNoRes

### CODE REVIEW

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 1]
```diff
@@ -321,6 +321,10 @@ sap.ui.define([
 			aResourcePath = [oResource.resourceType , "§" + oResource.expansion.identifier + "§"];
 			oResource = oResource.expansion.contains;
 		} else {
+			//generate a uuid id incase resource id is not present in the response
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:324]: [style] add whitespace after //

File: test/qunit/model/FHIRModel.unit.js [hunk 2]
```diff
@@ -1161,4 +1161,15 @@ sap.ui.define([
 		assert.strictEqual(oResource.id, "254", "Response with location without having / at the beginning gives proper resource object and doesnot throw error");
 	});
 
+	QUnit.test("Search Bundle Respone Shouldnot Throw Error If Resource Id is not present ", function(assert) {
+		var oJSONData = TestUtils.loadJSONFile("BundleWithoutResourceId");
+		var mResponseHeaders = {"etag" : "W/\"1\""};
+		this.oRequestHandle.setUrl("https://example.com/fhir/Patient?_count=0&total=accurate&_format=json");
+	    this.oRequestHandle.setRequest(TestUtils.createAjaxCallMock(mResponseHeaders));
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1168]: [style] add one more tab intend.

File: test/qunit/model/FHIRModel.unit.js [hunk 3]
```diff
@@ -1161,4 +1161,15 @@ sap.ui.define([
 		assert.strictEqual(oResource.id, "254", "Response with location without having / at the beginning gives proper resource object and doesnot throw error");
 	});
 
+	QUnit.test("Search Bundle Respone Shouldnot Throw Error If Resource Id is not present ", function(assert) {
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1164]: [style] "Search Bundle Response should not throw an error if ResourceId is not present"

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 1 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 2 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 3 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: src/sap/fhir/model/r4/FHIRModel.js [hunk 4]
```diff
@@ -321,6 +321,10 @@ sap.ui.define([
 			aResourcePath = [oResource.resourceType , "§" + oResource.expansion.identifier + "§"];
 			oResource = oResource.expansion.contains;
 		} else {
+			// generate a uuid id incase resource id is not present in the response
```
Comment (flovogt) [inline src/sap/fhir/model/r4/FHIRModel.js:324]: [style] "incase" add a missing whitespace between "in" and "case"

File: test/qunit/model/FHIRModel.integration.js [hunk 5]
```diff
@@ -435,7 +435,7 @@ sap.ui.define([
 		var mParameters = {
 			success : function(oData){
 				var oValidateResponse = TestUtils.loadJSONFile("Validate");
-				assert.deepEqual(oData, oValidateResponse, "Validate response was correct");
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.integration.js:438]: We should check here following things:

- the whole response is identical with the expected response only the resourceId is different
- the resourceId is has valid UUIDv4 value

File: test/qunit/model/FHIRModel.unit.js [hunk 6]
```diff
@@ -1161,4 +1161,15 @@ sap.ui.define([
...
 
+	QUnit.test("Search Bundle Response should not throw an error if ResourceId is not present", function(assert) {
+		var oJSONData = TestUtils.loadJSONFile("BundleWithoutResourceId");
+		var mResponseHeaders = { "etag": "W/\"1\"" };
+		this.oRequestHandle.setUrl("https://example.com/fhir/Patient?_count=0&total=accurate&_format=json");
+		this.oRequestHandle.setRequest(TestUtils.createAjaxCallMock(mResponseHeaders));
+		this.oFhirModel1._onSuccessfulRequest(this.oRequestHandle, oJSONData);
+		var sOperationOutcomePath = "/OperationOutcome";
+		var oOperationOutcome = this.oFhirModel1.getProperty(sOperationOutcomePath);
+		assert.strictEqual(Object.keys(oOperationOutcome).length, 1, "Bundle Response without id is parsed without throwing error");
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1172]: [style] "Bundle Response without resource id is parsed without throwing an error"

File: test/qunit/model/FHIRModel.unit.js [hunk 7]
```diff
@@ -1161,4 +1161,15 @@ sap.ui.define([
 		assert.strictEqual(oResource.id, "254", "Response with location without having / at the beginning gives proper resource object and doesnot throw error");
 	});
 
+	QUnit.test("Search Bundle Response should not throw an error if ResourceId is not present", function(assert) {
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1164]: [style] "Search Bundle Response should not throw an error if resource id is not present"

File: src/sap/fhir/model/r4/FHIRModel.js (hunk 4 above)
Comment (AmirthavalliG) [reply src/sap/fhir/model/r4/FHIRModel.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 6 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 7 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.integration.js (hunk 5 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.integration.js:438]: done

Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.integration.js:438]: done

File: test/qunit/model/FHIRModel.unit.js [hunk 8]
```diff
@@ -1161,4 +1161,15 @@ sap.ui.define([
 		assert.strictEqual(oResource.id, "254", "Response with location without having / at the beginning gives proper resource object and doesnot throw error");
 	});
 
+	QUnit.test("Search Bundle Response should not throw an error if resource id is not presen", function(assert) {
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1164]: missing "t" in word "present"

**Pull request merged**